orchagent/vxlanorch.cpp

```diff
@@ -725,48 +1387,281 @@ bool VxlanTunnelOrch::delOperation(const Request& request)
 
-    vxlan_tunnel_table_.erase(tunnel_name);
+    vtep_ptr->createDynamicDIPTunnel(remote_vtep, usr);
+    dip_tunnel = vtep_ptr->getDipTunnel(remote_vtep);
 
-    SWSS_LOG_NOTICE("Vxlan tunnel '%s' was removed", tunnel_name.c_str());
+    SWSS_LOG_NOTICE("diprefcnt for remote %s = %d",
+                     remote_vtep.c_str(), vtep_ptr->getDipTunnelRefCnt(remote_vtep));
+
+    if (!tunnel_orch->getTunnelPort(remote_vtep, tunport))
+    {
+      Port tunnelPort;
+      auto port_tunnel_name = "Port_EVPN_" + remote_vtep;
+      gPortsOrch->addTunnel(port_tunnel_name,dip_tunnel->getTunnelId(), false);
+      gPortsOrch->getPort(port_tunnel_name,tunnelPort);
+      gPortsOrch->addBridgePort(tunnelPort);
+    }
+
+    return true;
+}
+
+bool  VxlanTunnelOrch::delTunnelUser(const std::string remote_vtep, uint32_t vni_id, 
+                                    uint32_t vlan, tunnel_user_t usr,
+                                    sai_object_id_t vrf_id)
+{
+    if (TUNNEL_USER_MAC == usr) return true;
+
+    // If this is the last request to delete the tunnel.
+    auto tunnel_name = "EVPN_" + remote_vtep;
+
+    //if (isTunnelExists(tunnel_name))
+    {
+        auto port_tunnel_name = "Port_EVPN_" + remote_vtep;
+        EvpnNvoOrch* evpn_orch = gDirectory.get<EvpnNvoOrch*>();
+
+        auto vtep_ptr = evpn_orch->getEVPNVtep();
+
+        if (!vtep_ptr) 
+        {
+          SWSS_LOG_WARN("Unable to find VTEP. remote=%s vlan=%d usr=%d",
+                                             remote_vtep.c_str(), vlan, usr);
+          return true;
+        }
+
+        Port tunnelPort;
+        gPortsOrch->getPort(port_tunnel_name,tunnelPort);
+
+        if ((vtep_ptr->getDipTunnelRefCnt(remote_vtep) == 1) &&
+           tunnelPort.m_fdb_count == 0)
+        {
+          bool ret;
+
+          ret = gPortsOrch->removeBridgePort(tunnelPort);
+          if (!ret) 
+          {
+            SWSS_LOG_ERROR("Remove Bridge port failed for remote = %s fdbcount = %d", 
+                            remote_vtep.c_str(), tunnelPort.m_fdb_count);
+            return true;
+          }
+        
+          gPortsOrch->removeTunnel(tunnelPort);
+        }
+
+        vtep_ptr->deleteDynamicDIPTunnel(remote_vtep, usr);
+        SWSS_LOG_NOTICE("diprefcnt for remote %s = %d",
+                         remote_vtep.c_str(), vtep_ptr->getDipTunnelRefCnt(remote_vtep));
+
+        vtep_ptr->deletePendingSIPTunnel();
+
+    }
 
     return true;
 }
 
+void VxlanTunnelOrch::deleteTunnelPort(Port &tunnelPort)
+{
+    bool ret;
+    EvpnNvoOrch* evpn_orch = gDirectory.get<EvpnNvoOrch*>();
+    std::string remote_vtep;
+
+    auto vtep_ptr = evpn_orch->getEVPNVtep();
+
+    if (!vtep_ptr) 
+    {
+      SWSS_LOG_WARN("Unable to find VTEP. tunnelPort=%s",tunnelPort.m_alias.c_str());
+      return;
+    }
+
+    getTunnelDIPFromPort(tunnelPort, remote_vtep);
+
+    //If there are IMR/IP routes to the remote VTEP then ignore this call
+    if (vtep_ptr->getDipTunnelRefCnt(remote_vtep))
```

Comment:
getDipTunnelRefCnt > 1 condition. This method returns -1 if remote_vtep doesn't exist.